Sources/ProjectSpec/SupportedDestination.swift

```diff
@@ -40,6 +43,8 @@ extension SupportedDestination {
             return 3
         case .macCatalyst:
             return 4
+        case .watchOS:
+            return 5
```

Comment:
Hi @yonaskolb and @amatig,
I'm wondering why the priorities are currently in this order and where `watchOS` should be. Any thoughts?

Reply:
Adding test cases helped me realize this - `TARGETED_DEVICE_FAMILY` and `SUPPORTED_PLATFORMS` are ordered based on this priority. watchOS is `4`, meaning it needs to be between tvOS (`3`) and visionOS (`7`).